client/network/src/protocol/block_requests.rs

```diff
@@ -64,11 +67,24 @@ use void::{Void, unreachable};
 // Type alias for convenience.
 pub type Error = Box<dyn std::error::Error + 'static>;
 
+/// Event generated by the block requests behaviour.
+#[derive(Debug)]
+pub enum Event<B: Block> {
+	/// A response to a block request has arrived.
+	Response {
+		peer: PeerId,
+		/// The original request passed to `send_request`.
+		original_request: message::BlockRequest<B>,
+		response: message::BlockResponse<B>,
+	},
+}
+
 /// Configuration options for `BlockRequests`.
 #[derive(Debug, Clone)]
 pub struct Config {
 	max_block_data_response: u32,
```

Comment:
What happens if this is reached? How's it different from `max_response_len`?

Reply:
That's the maximum number of blocks that we respond with, by default 128. If a node requests 129 blocks, we will only answer with 128.